Assets/MixedRealityToolkit.SDK/Features/Utilities/Solvers/HandConstraint.cs

```diff
@@ -365,6 +365,23 @@ protected virtual IEnumerator ToggleCursor(bool visible, bool frameDelay = false
             }
         }
 
+        /// <summary>
+        /// Tells the SolverHandler to refresh its tracked object (hand)
+        /// </summary>
+        /// <remarks>
+        /// delayed 2 frames due to IMixedRealityHandJointService updating during "LateUpdate"
+        /// </remarks>
+        protected IEnumerator DelayedRefreshSolverHandlerTrackedObject()
```

Comment:
Yea we need a different solution. Waiting for two frames isn't reliable code.

@cameron-micka, FYI

Reply:
what's the alternative ?

1- (fast and easy) make DetachFromCurrentTrackedObject public or at least internal ?

2- (slow but better?) add some events in HandJointService and refactor it so that we can detect when a hand gets detected/lost + register to those events in HandConstraint.

3- other?

Reply:
I'll talk with @Cameron-Micka . SolverHandler should already be tracking if a hand is lost and can switch hands (with any solver like Surface Magnetism, etc). I think HandConstraint can be simplified but I need to confirm my understanding with Cameron.

Reply:
Following up on this, HandConstraint can be modified to be simpler and let SolverHandler handle the hands tracking.

HandConstraintPalmUp needs to still do some management because it activates only when palm is up

Reply:
what the plan ? does @Cameron-Micka handles the simplification ?
should i revert some commits here to only keep the events changes ?

Reply:
Sorry for the delay. I started the simplification in this branch 
https://github.com/Troy-Ferrell/MixedRealityToolkit-Unity/tree/users/trferrel/handconstraint-solverhandler

Reply:
ok, no problem.
feel free to close this PR when yours is up.